### PR TITLE
Add basic infrastructure using @pollyjs.

### PR DESCRIPTION
**Pull request by rwjblue**

### CODE REVIEW

File: src/commands/report-failure.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  test('creates an issue', async function() {
+    setupPolly('basic-test');
+
+    await reportFailure({
+      env: {
+        OWNER: 'malleatus',
+        REPO: 'nyx',
+        RUN_ID: '123456',
+        GITHUB_TOKEN: process.env.GITHUB_AUTH,
```
Comment (rwjblue) [inline src/commands/report-failure.test.ts]: Should probably not be committed 🤔

File: src/commands/report-failure.test.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  // ensure that the authorization token is not written to disk
+  saveRecording(recordingId: string, data: any) {
+    data.log.entries.forEach((entry: any) => {
+      entry.request.headers = entry.request.headers.filter((h: any) => h.name !== 'authorization');
+    });
+    // @ts-ignore
+    return super.saveRecording(recordingId, data);
+  }
+}
```
Comment (rwjblue) [inline src/commands/report-failure.test.ts]: It seems a bit unfortunate that we have to manually munge to "sanitize" the on disk output, but I think "it's fine" probably 🤔

File: package.json [hunk 3]
```diff
@@ -26,7 +26,14 @@
...
   },
   "devDependencies": {
+    "@pollyjs/adapter-node-http": "^4.0.3",
+    "@pollyjs/core": "^4.0.2",
+    "@pollyjs/persister-fs": "^4.0.2",
     "@types/jest": "^25.1.4",
+    "@types/pollyjs__adapter-node-http": "^2.0.0",
+    "@types/pollyjs__core": "^4.0.0",
+    "@types/pollyjs__persister-fs": "^2.0.0",
+    "@types/setup-polly-jest": "^0.5.0",
```
Comment (hjdivad) [inline package.json]: unused; remove

File: src/commands/report-failure.test.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import setupHardRejection from 'hard-rejection';
+
+class SanitizingPersister extends FSPersister {
+  static get id() {
+    return 'sanitizing-fs';
+  }
+
+  get options() {
+    return {
+      recordingsDir: path.resolve(__dirname, '__recordings__'),
```
Comment (hjdivad) [inline src/commands/report-failure.test.ts]: let's dotfile the recordings dir

Comment (hjdivad) [reply src/commands/report-failure.test.ts]: maybe `${projectRoot}/.recordings`

File: src/commands/report-failure.test.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    };
+  }
+
+  // ensure that the authorization token is not written to disk
+  saveRecording(recordingId: string, data: any) {
```
Comment (hjdivad) [inline src/commands/report-failure.test.ts]: let's at least leave a `//TODO: add type for data`

Comment (hjdivad) [reply src/commands/report-failure.test.ts]: actually better yet let's write an interface for what we use if we can't get an actual type

File: src/commands/report-failure.test.ts (hunk 2 above)
Comment (hjdivad) [reply src/commands/report-failure.test.ts]: patches welcome i'm sure

File: src/commands/report-failure.test.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    }
+  });
+
+  test('creates an issue', async function() {
+    setupPolly('basic-test');
+
+    await reportFailure({
+      env: {
+        OWNER: 'malleatus',
+        REPO: 'nyx',
```
Comment (hjdivad) [inline src/commands/report-failure.test.ts]: let's do this in nyx-example

File: src/commands/report-failure.test.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  afterEach(async () => {
+    if (polly) {
+      await polly.stop();
+    }
+  });
+
+  test('creates an issue', async function() {
+    setupPolly('basic-test');
+
+    await reportFailure({
```
Comment (hjdivad) [inline src/commands/report-failure.test.ts]: Assertions to add:

1. assert no issue initially
2. report failure
3. assert issue now exists and with expected content

**Comment by rwjblue** [comment]
Closed in favor of #28

**Pull request closed without merging**